synapse/types.py

```diff
@@ -67,6 +74,12 @@ class Collection(Iterable[T_co], Container[T_co], Sized):  # type: ignore
 JsonDict = Dict[str, Any]
 
 
+class SynapseReactor(
+    IReactorTCP, IReactorPluggableNameResolver, IReactorTime, IReactorCore
+):
+    """The interfaces necessary for Synapse to function."""
```

Comment:
Just for clarification, is this new type meant to be a collection of interfaces that we use as "okay, this type implements these interfaces"? I'm asking because i'm thinking about how this'd potentially interact with #8642 or other reactors.

Reply:
It is meant to collect the various reactor interfaces that Synapse expects, e.g. it expects to be able to speak TCP. It shouldn't have any impact on #8642, since if asyncioreactor doesn't implement these interfaces than it won't work!

Reply:
I've become lukewarm on asyncioreactor anyways, as in my research its proven to be a very thin wrapper around asyncio, with non-adequate integration with it (in my opinion).

(P.S. I cant seem to resolve this comment convo, but i consider it resolved, thanks for answering :+1:)